mmv1/templates/terraform/examples/network_connectivity_policy_basic.tf.erb

```diff
@@ -0,0 +1,34 @@
+resource "google_compute_network" "consumer_net" {
+  name                    = "<%= ctx[:vars]['consumer_network_name'] %>"
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "consumer_subnet" {
+  name          = "<%= ctx[:vars]['consumer_subnetwork_name'] %>"
+  ip_cidr_range = "10.0.0.0/16"
+  region        = "us-central1"
+  network       = google_compute_network.consumer_net.id
+}
+
+resource "google_compute_network" "producer_net" {
+  name                    = "<%= ctx[:vars]['producer_network_name'] %>"
+  auto_create_subnetworks = false
+}
+
+resource "google_compute_subnetwork" "producer_subnet" {
+  name          = "<%= ctx[:vars]['producer_subnetwork_name'] %>"
+  ip_cidr_range = "10.0.0.0/16"
+  region        = "us-central1"
+  network       = google_compute_network.producer_net.id
+}
+
+resource "google_network_connectivity_service_connection_policy" "default" {
+  name = "<%= ctx[:vars]['resource_name'] %>"
+  location = "us-central1"
+  service_class = "gcp-memorystore-redis"
```

Comment:
Can you briefly explain how this value is working? This will show up in the official documentation for this resource, and it will be used for testing in a number of different environments, so I'd like to understand if `gcp-memorystore-redis` is intended to be used in all of those places.

Reply:
Ohh, great, I just forgot that.
Yeah this should not be public. I've talked with a person for the serviceConnectionPolicy and mentioned that I need a way to make the tests work. So he gave me this resource to use. So, it's not for all those places, just here, temporarily while we don't support service_class.

How can I set this one as an env var?

Reply:
Sorry, I missed this comment on the prior pass.

Could you share any more about this value? Is it the name of an actual resource, or is it some sort of fake value? If it is an actual resource, is it publicly accessible (for anyone who chooses to run these tests)? If any sort of access is needed for it, we will need someone to make sure that access is set up for all of our internal environments.

Reply:
@roaks3 according to someone from the ServiceConnectionPolicies, this is a real resource, it's not a fake one, that it will be used while we don't have the serviceClass public available. It's not publicly accessible and I don't know how to make sure that all the envs will be added to have access. I will ask him to talk with you about that.

Reply:
Ok, sounds good. Depending on that conversation, we may want to hold this from being merged, or skip the tests right before we merge, but in the meantime:

- We will definitely want the docs to have some sort of example value. You can do this by adding a new field under `vars` that specifies the service class value that we would use in the docs (like `example-class`). Then you can also add a `test_vars_overrides` block with the value that will be used for tests only (you can search for some existing examples of this).
- We tend to avoid environment variables when possible, because we need to manage them centrally, and it would become overwhelming if they were needed by many teams/resources. Normally in cases like this, we would keep the test value as you have it (a hardcoded value), unless there is some sort of concern around doing that.

Reply:
Ohh, I got the idea.

I've added some more code to help with env_var.